### PR TITLE
1D implementations of median and count_nonzero

### LINKED ISSUE #3006: count_nonzero function
Replicate numpy.count_nonzero:
https://numpy.org/doc/stable/reference/generated/numpy.count_nonzero.html

### LINKED ISSUE #3007: median function
Replicate numpy.median function:
https://numpy.org/doc/stable/reference/generated/numpy.median.html

### PR DESCRIPTION
**Pull request by drculhane**
closes #3006 and closes #3007, median and count_nonzero, with 1D implementations of each.

### CODE REVIEW

File: arkouda/numeric.py [hunk 1]
```diff
@@ -2141,3 +2123,68 @@ def clip(
     if hi is not None:
         pda1 = where(pda1 > hi, hi, pda1)
     return pda1
+
+
+def median(pda):
+    """
+    Compute the median of a given array.  1d case only, for now.
+
```
Comment (ajpotts) [inline arkouda/numeric.py:2131]: numpy can handle boolean.  How hard would this be to add?  Can it be included or should it be a separate ticket?
```
In [29]: np.median(np.array([True, True, False]))
Out[29]: 1.0

```

File: arkouda/numeric.py [hunk 2]
```diff
@@ -2141,3 +2123,68 @@ def clip(
...
+    ----------
+    pda: pdarray
+        the input data, in pdarray form, numeric type
+
+    Returns
+    -------
+    an np.float64 value, which is the median of the entire pdarray
+
+    Examples
+    --------
```
Comment (ajpotts) [inline arkouda/numeric.py:2142]: Would you please write the examples as executable code, something like the following?

```python
    >>> import arkouda as ak
    >>> arkouda.connect()
    >>> pda = ak.array([0,4,7,8,1,3,5,2,-1])
    >>> ak.sort(pda)
    array([-1 0 1 2 3 4 5 7 8])

    >>> ak.median(pda)
    3

```

File: arkouda/numeric.py [hunk 3]
```diff
@@ -2141,3 +2123,68 @@ def clip(
...
+    Compute the median of a given array.  1d case only, for now.
+
+    Parameters
+    ----------
+    pda: pdarray
+        the input data, in pdarray form, numeric type
+
+    Returns
+    -------
+    an np.float64 value, which is the median of the entire pdarray
```
Comment (ajpotts) [inline arkouda/numeric.py:2139]: numpydoc format is:

```python
    Returns
    -------
    np.float86
        The median of the entire pdarray.
```

File: arkouda/numeric.py [hunk 4]
```diff
@@ -2141,3 +2123,68 @@ def clip(
...
+
+    Parameters
+    ----------
+    pda: pdarray
+        the input data, in pdarray form, numeric, bool, or str
+
+    Returns
+    -------
+    an np.int64 value, which is the nonzero count of the entire pdarray
+
```
Comment (ajpotts) [inline arkouda/numeric.py:2174]: Same docstring comments as above.

**Review by ajpotts — APPROVED** [review]

File: arkouda/numeric.py (hunk 1 above)
Comment (drculhane) [reply arkouda/numeric.py:2131]: ak.sort doesn't take bool, so we'd have to convert, but I don't see any reason not to do it.

File: arkouda/numeric.py [hunk 5]
```diff
@@ -2141,3 +2123,68 @@ def clip(
...
+    --------
+    pda = [0,4,7,8,1,3,5,2,-1] returns 9 (9 nonzero numerical elements)
+    pda = [False,True,False,True,False] returns 3 (3 True booleans)
+    pda = ["hello","","there"] returns 2 (2 non-empty strings)
+
+    """
+
+    #  Handle different data types.
+
+    if (pda.dtype == np.int64) or (pda.dtype == np.float64):
```
[the remark this answers is not in the document]
Comment (drculhane) [reply arkouda/numeric.py:2185]: Done.

File: arkouda/numeric.py (hunk 4 above)
Comment (drculhane) [reply arkouda/numeric.py:2174]: Done.

File: arkouda/numeric.py (hunk 3 above)
Comment (drculhane) [reply arkouda/numeric.py:2139]: Done.

File: arkouda/numeric.py (hunk 2 above)
Comment (drculhane) [reply arkouda/numeric.py:2142]: Done.

**Review by stress-tess — APPROVED** [review]
Looks good to me! nice work @drculhane

**Review by jaketrookman — APPROVED** [review]

**Pull request merged**